src/esp/sensor/Sensor.cpp

```diff
@@ -11,29 +11,19 @@
 namespace esp {
 namespace sensor {
 
+SensorSpec::~SensorSpec() {}
+
 Sensor::Sensor(scene::SceneNode& node, SensorSpec::ptr spec)
-    : Magnum::SceneGraph::AbstractFeature3D{node}, spec_(std::move(spec)) {
+    : Magnum::SceneGraph::AbstractFeature3D{node}, spec_(spec) {
```

Comment:
What is the reason that you removed the `std::move`?

Reply:
I noticed that at runtime, keeping `std::move(spec)` would cause VisualSensor's spec to be nullptr, while a separate spec existed in memory. I removed it and it seemed to work. Could it be due to how in `VisualSensor.h`, I have `VisualSensorSpec::ptr spec_ = std::dynamic_pointer_cast<VisualSensorSpec>(spec_);` to make sure that methods can access parameters in VisualSensorSpec but not SensorSpec?